snuba/web/views.py

```diff
@@ -269,46 +264,60 @@ def dataset_query(dataset: Dataset, body, timer: Timer) -> Response:
     assert http_request.method == "POST"
     ensure_not_internal(dataset)
     ensure_table_exists(dataset)
-    return format_result(
-        run_query(
-            dataset,
-            validate_request_content(
-                body,
-                RequestSchema.build_with_extensions(
-                    dataset.get_extensions(), HTTPRequestSettings
-                ),
-                timer,
-                dataset,
-                http_request.referrer,
-            ),
-            timer,
-        )
-    )
 
+    request = validate_request_content(
+        body,
+        RequestSchema.build_with_extensions(
+            dataset.get_extensions(), HTTPRequestSettings
+        ),
+        timer,
+        dataset,
+        http_request.referrer,
+    )
 
-def run_query(dataset: Dataset, request: Request, timer: Timer) -> WebQueryResult:
     try:
         result = parse_and_run_query(dataset, request, timer)
-        payload = {**result.result, "timing": timer.for_json()}
-        if settings.STATS_IN_RESPONSE or request.settings.get_debug():
-            payload.update(result.extra)
-        return WebQueryResult(payload, 200)
-    except RawQueryException as e:
-        return WebQueryResult(
-            {
-                "error": {"type": e.err_type, "message": e.message, **e.meta},
-                "sql": e.sql,
-                "stats": e.stats,
-                "timing": timer.for_json(),
-            },
-            429 if e.err_type == "rate-limited" else 500,
+    except RawQueryException as exception:
```

Comment:
I considered making this an error handler like we do with some other exception types like [`BadRequest`](https://github.com/getsentry/snuba/blob/master/snuba/web/views.py#L102-L131) but this seemed clearer to me.